AffirmSDK/AffirmAsLowAsData.m

```diff
@@ -92,22 +42,16 @@ + (void)sendPricingRequest:(NSDecimalNumber *)amount
 
 + (void)getALAPromoTemplateForAmount:(NSDecimalNumber *)amount
                              promoID:(NSString *)promoID
-                            callback:(void (^)(NSString *promoTemplate, BOOL promoPrequalEnabled, NSError *error, BOOL success))callback {
-    [self getPromoDetailsForId:promoID amount:amount callback:^(AffirmLoanTerm *minLoanTerm, BOOL promoPrequalEnabled, NSError *error, BOOL success) {
-        NSString *promoDefaultTemplate = minLoanTerm.defaultMessage ? minLoanTerm.defaultMessage : defaultALATemplate;
+                            callback:(void (^)(NSString *ala, BOOL showPrequal, NSError *error, BOOL success))callback {
+    [self getPromoDetailsForId:promoID amount:amount callback:^(NSDictionary *result, NSError *error, BOOL success) {
         if (success) {
-            [self sendPricingRequest:amount loanTerm:minLoanTerm callback:^(AffirmPricing *pricing, NSError *error, BOOL pricingSuccess) {
-                NSString *template = promoDefaultTemplate;
-                if (pricingSuccess && [minLoanTerm.pricingTemplate isKindOfClass:[NSString class]]) {
-                    template = [minLoanTerm.pricingTemplate stringByReplacingOccurrencesOfString:@"{payment}" withString:[NSString stringWithFormat:@"$%@", pricing.paymentString]];
-                    template = [template stringByReplacingOccurrencesOfString:@"{lowest_apr}" withString:[NSString stringWithFormat:@"%@", @(minLoanTerm.apr.floatValue * 100)]];
-                }
-
-                template = [template stringByReplacingOccurrencesOfString:@"{affirm_logo}" withString:@"Affirm"];
-                callback(template, promoPrequalEnabled, error, pricingSuccess);
-            }];
+            NSString *ala = [result[@"promo"][@"ala"] copy];
+            ala = [ala stringByReplacingOccurrencesOfString:@"{affirm_logo}" withString:@"Affirm"];
+            
+            NSString *style = [result[@"promo"][@"config"][@"promo_style"] copy];
+            callback(ala, [style isEqualToString:@"fast"], error, success);
```

Comment:
I think it would be more clear to have `BOOL showPrequal = [style isEqualToString:@"fast"]` on a separate line than directly in the call to `callback`.